genai-perf/genai_perf/inputs/retrievers/synthetic_prompt_generator.py

```diff
@@ -155,17 +161,18 @@ def _generate_prompt_with_token_reuse(
         Returns:
             str: A synthetic prompt as a string.
         """
-        final_prompt = []
+        final_prompt: List[int] = []
         size_to_use = block_size
         for index, hash_index in enumerate(prompt_hash_list):
             if index == len(prompt_hash_list) - 1:
                 size_to_use = num_tokens - (index * block_size)
             if hash_index not in cls._cache:
-                prompt = cls._generate_prompt(tokenizer, size_to_use)
+                prompt = cls._generate_prompt(tokenizer, size_to_use, decode=False)
```

Comment:
Can we have a comment explaining this hack? It will be super helpful when someone else has to revisit this block of code.

Reply:
I added a comment. Good call out @nv-hwoo